### PR TITLE
HAAR 522: Ensure creating user and changing name allow apostrophe in last name

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserAccountResourceIntTest.kt [hunk 1]
```diff
@@ -50,7 +91,7 @@ class UserAccountResourceIntTest : IntegrationTestBase() {
             CreateLocalAdminUserRequest(
               username = "laauser1",
               firstName = "Laa",
-              lastName = "User",
+              lastName = "U'ser",
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserAccountResourceIntTest.kt:94]: Create user tests pulled into inner class to and the create tests (for local/admin/general user) now add in an apostrophe for the "successful" create

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserAccountResourceIntTest.kt [hunk 2]
```diff
@@ -73,7 +114,43 @@ class UserAccountResourceIntTest : IntegrationTestBase() {
...
+          "primaryEmail": "testgen@test.com",
+          "accountType": "GENERAL"
+          }
+          """
+        )
+    }
+  }
+
+  @Nested
+  @DisplayName("POST /users/link...")
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserAccountResourceIntTest.kt:150]: just separated create and link tests

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserManagementResourceIntTest.kt [hunk 3]
```diff
@@ -414,7 +414,26 @@ class UserManagementResourceIntTest : IntegrationTestBase() {
         .expectStatus().is4xxClientError
         .expectBody()
         .jsonPath("userMessage")
-        .isEqualTo("Validation failure: Last name must consist of alphabetical characters only and a max 35 chars")
+        .isEqualTo("Validation failure: Last name must consist of alphabetical characters or an apostrophe only and a max 35 chars")
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserManagementResourceIntTest.kt:417]: Wondered if the creates should also do this kind of test? to test error messages ?

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**